LoopKitUI/Views/GuidePage.swift

```diff
@@ -8,34 +8,54 @@
 
 import SwiftUI
 
-public struct GuidePage<Content>: View where Content: View {
-    
-    let content: () -> Content
-    
-    public init(@ViewBuilder content: @escaping () -> Content) {
-        self.content = content
+public struct GuidePage<Content, ActionAreaContent>: View where Content: View, ActionAreaContent: View {
+    let content: Content
+    let actionAreaContent: ActionAreaContent
+
+    @Environment(\.horizontalSizeClass) var horizontalSizeClass
+
+    public init(@ViewBuilder content: @escaping () -> Content,
+         @ViewBuilder actionAreaContent: @escaping () -> ActionAreaContent)
+    {
+        self.content = content()
+        self.actionAreaContent = actionAreaContent()
     }
-    
+
     public var body: some View {
-        GeometryReader { geometry in
-            ScrollView {
-                VStack {
-                    self.content()
+        VStack(spacing: 0) {
+            List {
+                if self.horizontalSizeClass == .compact {
+                    Section(header: EmptyView(), footer: EmptyView()) {
+                        self.content
+                    }
+                } else {
+                    self.content
                 }
-                .padding()
-                .frame(minHeight: geometry.size.height)
-                .background(Color(UIColor.systemGroupedBackground))
             }
+            .environment(\.horizontalSizeClass, self.horizontalSizeClass)
```

Comment:
I was hoping `List` would be included. I think this will overcome some UI weirdness I have noticed, most obvious the scrolling of the navigation title.